src/routes/(app)/checkout/+page.svelte

```diff
@@ -33,8 +33,14 @@
 	let onLocation = data.defaultOnLocation;
 	$: offerDeliveryFees = onLocation;
 	let addDiscount = false;
+	let offerOrder = false;
 	let discountAmount = 0;
 	let discountType: DiscountType;
+	$: if (offerOrder) {
+		discountType = 'percentage';
+		discountAmount = 100;
+	}
```

Comment:
The problem is if the user changes the discount amount, the `offerOrder` checkbox stays ticked, that's why I added the second `if` in the comment

Reply:
The second if was giving this error `Cyclical dependency detected: discountType → offerOrder → discountType`

Reply:
you can try reordering the elements

or find another solution

Reply:
When offer order is checked you can't update discountAmount and discountType